contracts/zaps/ZapTrancheLp.sol

```diff
@@ -0,0 +1,277 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.0;
+pragma experimental ABIEncoderV2;
+
+import "../libraries/Authorizable.sol";
+import "../interfaces/IERC20Permit.sol";
+import "../interfaces/ITranche.sol";
+import "../interfaces/IWrappedPosition.sol";
+import "../interfaces/IInterestToken.sol";
+
+interface IPool {
+    function bond() external view returns (address);
+}
+
+interface IAsset {}
+
+interface IVault {
+    function exitPool(
+        bytes32 poolId,
+        address sender,
+        address payable recipient,
+        ExitPoolRequest memory request
+    ) external;
+
+    struct ExitPoolRequest {
+        IAsset[] assets;
+        uint256[] minAmountsOut;
+        bytes userData;
+        bool toInternalBalance;
+    }
+
+    function joinPool(
+        bytes32 poolId,
+        address sender,
+        address recipient,
+        JoinPoolRequest memory request
+    ) external payable;
+
+    struct JoinPoolRequest {
+        IAsset[] assets;
+        uint256[] maxAmountsIn;
+        bytes userData;
+        bool fromInternalBalance;
+    }
+}
+
+interface ITrancheExt is ITranche {
+    function position() external view returns (IWrappedPosition);
+
+    function underlying() external view returns (IERC20);
+}
+
+contract ZapTrancheLp is Authorizable {
+    // Store the accessibility state of the contract
+    bool public isFrozen = false;
+
+    // The balancer Vault contract
+    IVault internal _vault;
+
+    constructor(address __vault) Authorizable() {
+        _authorize(msg.sender);
+        _vault = IVault(__vault);
+    }
+
+    /// @dev Requires that the contract is not frozen
+    modifier notFrozen() {
+        require(!isFrozen, "Contract frozen");
+        _;
+    }
+
+    /// @dev Allows an authorized address to freeze or unfreeze this contract
+    /// @param _newState True for frozen and false for unfrozen
+    function setIsFrozen(bool _newState) external onlyAuthorized() {
+        isFrozen = _newState;
+    }
+
+    // Memory encoding of the permit data
+    struct PermitData {
+        IERC20Permit tokenContract;
+        address who;
+        uint256 amount;
+        uint256 expiration;
+        bytes32 r;
+        bytes32 s;
+        uint8 v;
+    }
+
+    /// @dev Takes the input permit calls and executes them
+    /// @param data The array which encodes the set of permit calls to make
+    modifier preApproval(PermitData[] memory data) {
+        // If permit calls are provided we make try to make them
```

Comment:
I think just a small typo in this comment line
`// If permit calls are provided we try to make them`